src/auth/User.h

```diff
@@ -66,10 +66,15 @@ class User : public RefCountable
     const SBuf userKey() const {return userKey_;}
 
     /**
-     * How long these credentials are still valid for.
-     * Negative numbers means already expired.
+     * Credentials are not permitted to be re-used from a
+     * credentials cache unless the authentication scheme
+     * defines a way to determine a TTL with which to bound
+     * the scope of their valid re-use.
```

Comment:
Expired credentials reuse is also not permitted, right? Please add that critical information to the method description. Without that restriction, the method API does not really work because the method caller cannot distinguish "scheme did not define ..." cases from "credentials expired" cases.